benchmarks/fastrnns/factory.py

```diff
@@ -372,10 +372,16 @@ def dynamic_rnn(input, hidden, wih, whh, bih, bhh):
         # type: (Tensor, Tuple[Tensor, Tensor], Tensor, Tensor, Tensor, Tensor) -> Tuple[Tensor, Tuple[Tensor, Tensor]]
         hx, cx = hidden
         outputs = []
-        inputs = torch.matmul(input, wih.t()).unbind(0)
+        inpSize = input.size()
+        # add bias for all timesteps instead of going step-by-step, results in a single reduction kernel in the backward
+        # FIXME matmul(x,y) + bias currently goes through jit AD, and backward formula in AD is not optimized for this
+        # case. Workaround with mm and views.
+        inpSize = input.size()
+        inputs = torch.mm(input.view(-1, inpSize[2]), wih.t()) + bih
+        inputs = inputs.view(inpSize[0], inpSize[1], -1).unbind(0)
```

Comment:
Honestly I think we should add a new benchmark to see the effect of this change. We've been using this one for quite a while, so I'd rather keep its meaning consistent with what people expect

Reply:
The effect is on the order of couple percent, I'll add a separate benchmark.

Reply:
Actually, I'm getting crazy big (7%) improvement from adding bias on my current system, but it's very system-dependent. 
```
root@7a3abf660096:/workspace/ALL/pytorch_upstream/benchmarks# python -m benchmarks.fastrnns.bench --group rnns --inputSize 1024 --hiddenSize 1024 --rnns jit_premul jit_premul_bias jit cudnn --nloops 100
Namespace(cnns=None, device='cuda', group=['rnns'], hiddenSize=1024, inputSize=1024, miniBatch=64, nloops=100, numLayers=1, print_json=False, rnns=['jit_premul', 'jit_premul_bias', 'jit', 'cudnn'], sep=' ', seqLength=100, variable_lstms=False, warmup=10)
Benchmarking LSTMs...
            name          avg_fwd          std_fwd          avg_bwd          std_bwd
      jit_premul            10.52          0.02504             21.2            1.051
 jit_premul_bias             10.7          0.04063            19.62           0.2769
             jit            11.49          0.02089            20.96           0.2493
           cudnn            9.815          0.04521            18.98          0.09339
```